views/reports/monthly_report.php

```diff
@@ -1,1248 +1,1985 @@
+		
+		#Caucasian Queries
+		$query = $pMonthQuery . " AND $caucWhere ";
+		$caucMonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $caucWhere ";
+		$caucNewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$caucDupRes = $caucMonthRes - $caucNewRes;
+		$query = $pYearQuery . " AND $caucWhere ";
+		$caucYearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		
+		#African American Queries
+		$query = $pMonthQuery . " AND $afrAmWhere ";
+		$afAmMonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $afrAmWhere ";
+		$afAmNewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$afAmDupRes = $afAmMonthRes - $afAmNewRes;
+		$query = $pYearQuery . " AND $afrAmWhere ";
+		$afAmYearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		
+		#Multi-Racial
+		$query = $pMonthQuery . " AND $multRacWhere ";
+		$multRacMonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $multRacWhere ";
+		$multRacNewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$multRacDupRes = $multRacMonthRes - $multRacNewRes;
+		$query = $pYearQuery . " AND $multRacWhere ";
+		$multRacYearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		
+		#Latino
+		$query = $pMonthQuery . " AND $latWhere ";
+		$latMonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $latWhere ";
+		$latNewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$latDupRes = $latMonthRes - $latNewRes;
+		$query = $pYearQuery . " AND $latWhere ";
+		$latYearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		
+		#Pacific Islander
+		$query = $pMonthQuery . " AND $pacIslWhere ";
+		$pacIslMonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $pacIslWhere ";
+		$pacIslNewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$pacIslDupRes = $pacIslMonthRes - $pacIslNewRes;
+		$query = $pYearQuery . " AND $pacIslWhere ";
+		$pacIslYearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+
+		#Native American
+		$query = $pMonthQuery . " AND $natAmWhere ";
+		$natAmMonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $natAmWhere ";
+		$natAmNewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$natAmDupRes = $natAmMonthRes - $natAmNewRes;
+		$query = $pYearQuery . " AND $natAmWhere ";
+		$natAmYearRes = pg_fetch_result($db->query($query ,[]), 0, 0);	
+
+		#Other Races
+		$query = $pMonthQuery . " AND $otherRacWhere ";
+		$otherRacMonthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$query = $pNewQuery . " AND $otherRacWhere ";
+		$otherRacNewRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		$otherRacDupRes = $otherRacMonthRes - $otherRacNewRes;
+		$query = $pYearQuery . " AND $otherRacWhere ";
+		$otherRacYearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
+		
+		#Zip Codes
+		$query = $pMonthQuery . " AND zipcode = 12501 ";
```

Comment:
100% make a generic function for this. It's all the same except a couple of things.

Reply:
```php
$zipQuery = function($zip, &$monthRes, &$newRes, &$dupRes, &$yearRes)
        use ($db, $pMonthQuery, $pNewQuery, $pYearQuery) {
    $query = $pMonthQuery . " AND zipcode = $zip ";
    $monthRes = pg_fetch_result($db->query($query ,[]), 0, 0);
    $query = $pNewQuery . " AND zipcode = $zip ";
    $newRes = pg_fetch_result($db->query($query ,[]), 0, 0);
    $dupRes = $monthRes - $newRes;
    $query = $pYearQuery . " AND zipcode = $zip ";
    $yearRes = pg_fetch_result($db->query($query ,[]), 0, 0);
};

#Zip Codes
$zipQuery(12501, $_12501MonthRes, $_12501NewRes, $_12501DupRes, $_12501YearRes);
$zipQuery(12504, $_12504MonthRes, $_12504NewRes, $_12504DupRes, $_12504YearRes);
```

Reply:
Actually just use the function from the race.
```php
#Zip Codes
$queryBuild(" AND zipcode = 12501", $_12501MonthRes, $_12501NewRes, $_12501DupRes, $_12501YearRes);
$queryBuild(" AND zipcode = 12504", $_12504MonthRes, $_12504NewRes, $_12504DupRes, $_12504YearRes);
```